internal/apis/certmanager/v1alpha2/types_certificate.go

```diff
@@ -332,8 +332,32 @@ type PKCS12Keystore struct {
 	// PasswordSecretRef is a reference to a key in a Secret resource
 	// containing the password used to encrypt the PKCS12 keystore.
 	PasswordSecretRef cmmeta.SecretKeySelector `json:"passwordSecretRef"`
+
+	// Algorithm is the encryption algorithm used to create the PKCS12 keystore.
+	// Default value is `RC2` for backward compatibility.
+	//
+	// If provided, allowed values are:
+	// `RC2`: Deprecated. Not supported by default in OpenSSL 3 or Java 20.
+	// `DES3`: Less secure, used for maximal compatibility.
+	// `SHA256`: Preferred for security, used when indicated by policy. (PEM format also stored in Secret.)
+	// +optional
+	Algorithm PKCS12Algorithm `json:"algorithm,omitempty"`
 }
 
+// +kubebuilder:validation:Enum="RC2-40-CBC:HMAC-SHA-1";"AES-256-CBC:HMAC-SHA-2"
```

Comment:
These enums still have to be updated.
Also, did you check that this is validated? Otherwise, we might have to add a check in the validation webhook logic.